[1단계 - HTTP 서버 구현하기] 띠용(주용은) 미션 제출합니다

jye326 opened the pull request.

jye326 committed:
fix: 빌드 환경 수정

jye326 committed:
feat: FileTest 작성

jye326 committed:
feat: IOStreamTest 작성

jye326 committed:
feat: IOStreamTest 작성

jye326 committed:
빌드 설정 변경

jye326 committed:
fix: 테스트 수정

- 올바른 검증을 위해 하드코딩 된 값을 수정하였습니다.

jye326 committed:
feat: step1 기능 추가

1. GET /index.html 응답하기
2. CSS 지원하기
3. Query String 파싱

jye326 commented:
@coderabbitai review

Instructions 참고해서 코드 리뷰 진행해줘

jye326 committed:
refactor: process 메서드 리팩터링

- 메서드 분리
- 에러메시지 상수화
- 유저 확인 메서드 은닉

jye326 commented:
@coderabbitai review

수정사항 반영했으니 Instructions 참고해서 코드 리뷰 진행해줘

jye326 committed:
feat: process 메서드 내부 검증 로직 추가

- uri 검증 로직 추가

jye326 committed:
feat: 컨텐츠 타입 추출 메서드 안정화

jye326 committed:
refactor: 로그인 핸들 기능 메서드 분리

jye326 commented:
@coderabbitad 이 정도면 충분한 것 같지 않니

minjae8563 commented on study/src/test/java/study/FileTest.java:
```diff
@@ -27,8 +31,7 @@ class FileTest {
 
-        // todo
-        final String actual = "";
+        final String actual = getClass().getClassLoader().getResource(fileName).toString();
```
경로를 찾는 테스트니까 `toString()` 보다 `getPath()` 어떠신가요? 😄

jye326 replied:
동의합니다!

minjae8563 commented on study/src/test/java/study/IOStreamTest.java:
```diff
@@ -90,12 +91,15 @@ class OutputStream_학습_테스트 {
              * try-with-resources를 사용한다.
              * java 9 이상에서는 변수를 try-with-resources로 처리할 수 있다.
              */
+            try(outputStream){
```
👍

minjae8563 replied:
👍

minjae8563 commented on study/src/test/java/study/FileTest.java:
```diff
@@ -27,8 +31,7 @@ class FileTest {
 
-        // todo
-        final String actual = "";
+        final String actual = getClass().getClassLoader().getResource(fileName).toString();
```
경로 찾는 테스트이므로 `toString()` 보다는 `getPath()` 어떠신가요😄

minjae8563 commented on study/src/test/java/study/IOStreamTest.java:
```diff
@@ -169,13 +170,14 @@ class FilterStream_학습_테스트 {
 
+            final byte[] actual = bufferedInputStream.readAllBytes();
+            inputStream.close();
+            bufferedInputStream.close();
```
`try-with-resources` 써도 깔끔할 것 같아요 😄

jye326 replied:
그렇네요! 적용했습니다!

minjae8563 commented on tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java:
```diff
@@ -28,20 +41,89 @@ public void run() {
     public void process(final Socket connection) {
         try (final var inputStream = connection.getInputStream();
              final var outputStream = connection.getOutputStream()) {
+            BufferedReader br = new BufferedReader(new InputStreamReader(inputStream));
```
BufferedReader 도 `try-with-resources` 에 넣어주는 건 어떤가요 😄

jye326 replied:
좋네요! 반영했습니다!

minjae8563 commented on tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java:
```diff
@@ -28,20 +41,89 @@ public void run() {
+        if (parts.length < 3) {
+            throw new IllegalArgumentException(INVALID_HTTP_REQUEST_FORMAT.getMessage());
+        }
+        return parts[1];
```
parts[1] 도 변수로 빼주면 좋을 것 같아요 😄

minjae8563 commented on tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java:
```diff
@@ -28,20 +41,89 @@ public void run() {
+        final var responseBody = getResponseBodyFromStaticResource(uri);
+        final var response = String.join("\r\n",
+                "HTTP/1.1 200 OK ",
+                "Content-Type: text/" + contentType + ";charset=utf-8 ",
```
`Files.probeContentType(path)` 도 있더라고요 😄

jye326 replied:
이런 게 있었군요! `getConttentType`에 적용했습니다!

minjae8563 reviewed (commented):
고생하셨습니다~!

minjae8563 reviewed (approved):
다음 미션 진행하셔도 좋을 것 같아요~ 위에 리뷰에서 approve 못 눌러서 다시 눌러요 ~!

minjae8563 commented:
<img width="868" height="245" alt="스크린샷 2025-09-05 오후 8 18 02" src="https://github.com/user-attachments/assets/ddfe92a9-3fb5-4600-85ca-a734804825f1" />

`localhost:8080` 으로 접속 시 위와 같은 예외가 발생합니다.

<img width="1074" height="140" alt="스크린샷 2025-09-05 오후 8 18 29" src="https://github.com/user-attachments/assets/84f729df-be80-4b91-b37a-412974732074" />

`localhost:8080/index.html` 로 접속 시 위와 같은 예외가 발생합니다.

minjae8563 commented on tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java:
```diff
@@ -28,20 +41,89 @@ public void run() {
+    }
 
-            final var responseBody = "Hello world!";
+    private static String parseUri(BufferedReader br) throws IOException {
```
`static` 일 필요 없을 것 같아요~😄

jye326 replied:
인정합니다~

minjae8563 commented on tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java:
```diff
@@ -28,20 +41,89 @@ public void run() {
-                    "Content-Length: " + responseBody.getBytes().length + " ",
-                    "",
-                    responseBody);
+    private static void handleLogin(String uri) {
```
`static` 일 필요 없을 것 같아요~😄

minjae8563 replied:
해당 클래스에 static 메서드가 여러 개 보이는 것 같아요. 같은 리뷰 남기겠습니다~

jye326 committed:
refactor: 테스트 리팩터링

jye326 committed:
refactor: Http11Processor 리팩터링

- 메서드 파라미터 및 리턴 타입 개선
- `Path`와 `Files` 유틸 사용으로 리소스 처리 로직 최적화
- 정적 메서드를 인스턴스 메서드로 변경 및 코드 가독성 향상
- Http 응답 생성 로직 분리 및 메서드화

jye326 commented:
오류 수정 및 리팩터링 완료했습니다! 머지하겠습니다~
